File: Implemented Funs Using C++/String Reverse.cpp
#include<iostream>
#include<string>
#include<algorithm>
#include<stack>

using namespace std;

#define ll long long 

// Implemented Function Will Reverse The String
// s : The string that will be reversed

//The First Way Using swap
void reversStringWay1(string s) {
    for (int i = 0; i < (s.size()) / 2; i++) {
        // swap(s[i], s[s.size() - i - 1]);
        char x = s[i];
        s[i] = s[s.size() - i - 1];
        s[s.size() - i - 1] = x;

    }
    cout << s << endl;
}


//The Second Way Using Stack
void reversStringWay2(string s) {
    
    stack<char>x;
    for (int i = 0; i < s.size(); i++)
        x.push(s[i]);

    for (int i = 0; !x.empty(); i--) {
        cout << x.top();
        x.pop();
    }
    cout << endl;
}

//The Thired Way Using Array
void reversStringWay3(string s) {

    string x;
    
    for (int i = (s.size()) - 1; i >= 0; i--)   x += s[i];

    cout << x << endl;
}

int main()
{
    string s;  cin >> s;
    reversStringWay1(s);
    reversStringWay2(s);
    reversStringWay3(s);
      
    return 0;
}

File: Implemented Funs Using JS/String Reverse.js
// Implemented Function Will Reverse The String
// s : The string that will be reversed

//The First Way Using swap
function stringReverseWay1(s) {
  let array = s.split("");

  for (let i = 0; i < array.length / 2; i++) {
    let temp = array[i];
    array[i] = array[array.length - i - 1];
    array[array.length - i - 1] = temp;
  }

  console.log(array.join(""));
}

//The Second Way Using Empty String
function stringReverseWay2(s) {
  let reversString = "";
  for (let i = s.length - 1; i >= 0; i--) {
    reversString += s[i];
  }
  console.log(reversString);
}

//The Thired Way Using Array
function stringReverseWay3(s) {
  let result = [];
  for (let i = 0; i < s.length; i++) {
    result.unshift(s[i]);
  }
  console.log(result.join(""));
}

stringReverseWay1("Migo");

stringReverseWay2("Migo");

stringReverseWay3("Migo");
